src/ol/controls/RoutingControl.js

```diff
@@ -0,0 +1,359 @@
+        }
+
+        // Get the closest point from viaPoints to original route
+        const viaPointClosestPoints = this.viaPoints.map((viaPoint) => {
+          const snapPoint = oldRoute.getGeometry().getClosestPoint(viaPoint);
+          return snapPoint;
+        });
+
+        // Calculate the distance from the click point to each segment
+        let distanceToSegment;
+        for (let i = 0; i < viaPointClosestPoints.length - 1; i += 1) {
+          // Create a segment for each pair of (closest) viaPoints using turf lineSlice
+          const turfLineSegment = lineSlice(
+            createTurfPoint(viaPointClosestPoints[i]),
+            createTurfPoint(viaPointClosestPoints[i + 1]),
+            createTurfLine(oldRoute.getGeometry().getCoordinates()),
+          );
+
+          /* Create an ol LineString from segment and get the distance between
+           * the click point and the closest point on segment
+           */
+          const segment = new LineString(turfLineSegment.geometry.coordinates);
+          const distanceToClick = getDistance(
+            segment.getClosestPoint(this.initialRouteDrag.coordinate),
+            this.initialRouteDrag.coordinate,
+          );
+
+          // Return segment index with smallest distance to click point
+          if (!distanceToSegment || distanceToClick < distanceToSegment) {
+            distanceToSegment = distanceToClick;
+            newViaIndex = i + 1;
+          }
+        }
+
+        // Insert new viaPoint at given index
+        return this.addViaPoint(e.mapBrowserEvent.coordinate, newViaIndex);
+      });
+
+      // Add control ressources to map
+      this.map.addLayer(this.routingLayer);
+      this.onMapClickKey = this.map.on('singleclick', (e) => {
+        const feats = e.target.getFeaturesAtPixel(e.pixel);
+        const viaPoint = feats.find(
+          (feat) =>
+            feat.getGeometry() instanceof Point &&
+            feat.get('index') !== undefined,
+        );
+
+        if (viaPoint) {
+          // Remove existing viaPoint on click and abort viaPoint add
+          this.removeViaPoint(viaPoint.get('index'));
+          return;
+        }
+
+        this.addViaPoint(e.coordinate);
+      });
+
+      this.map.addInteraction(this.modify);
+    }
+  }
+
+  addViaPoint(coordinate, index = this.viaPoints.length, overwrite = 0) {
+    /* Add/Insert/Overwrite viapoint and redraw route */
+    this.viaPoints.splice(index, overwrite, coordinate);
+    this.drawRoute(this.viaPoints);
+  }
+
+  removeViaPoint(index = this.viaPoints.length - 1) {
+    /* Remove viapoint and redraw route */
+    if (this.viaPoints.length && this.viaPoints[index]) {
+      this.viaPoints.splice(index, 1);
+    }
+    this.drawRoute(this.viaPoints);
+  }
+
+  getViaPoints() {
+    /* Return array of viaPoints */
+    return this.viaPoints;
+  }
+
+  setMot(motString) {
+    /* Return array of viaPoints */
+    this.mot = motString;
+    this.drawRoute(this.viaPoints);
+  }
```

Comment:
Use a defineProperties method and define the setter for this property see common/Control for an example.

Reply:
Moved to RoutingAPI and using setter/getter methods